lib/video/screenRecording/desktop/firefoxrecorder.js

```diff
@@ -0,0 +1,161 @@
+const execa = require('execa');
+const del = require('del');
+
+const unlink = util.promisify(fs.unlink);
+
+async function createTempFile() {
+  const mkdtemp = util.promisify(fs.mkdtemp);
+  const dir = await mkdtemp(path.join(os.tmpdir(), 'browsertime-'));
+  return path.join(dir, 'winrecorder.mp4');
+}
+
+async function enableWindowRecorder(enable, driver) {
+  if (enable) {
+    enable = '1';
+  } else {
+    enable = '0';
+  }
+  const oldContext = driver.getContext();
+  await driver.setContext('chrome');
+  const script = 'windowUtils.setCompositionRecording(' + enable + ');';
+  await driver.executeScript(script);
+  return driver.setContext(oldContext);
+}
+
+function findRecordingDirectory() {
+  let closest_mtime = 0;
+  let directory = undefined;
+
+  fs.readdirSync('.').forEach(file => {
+    if (file.startsWith('windowrecording-')) {
+      let mtime = fs.statSync(file).mtime;
+      if (mtime > closest_mtime) {
+        closest_mtime = mtime;
+        directory = file;
+      }
+    }
+  });
+
+  log.debug('Using window recording directory: ' + directory);
+  return directory;
+}
+
+async function pollDirectory(directory) {
+  // We need to spin wait here until the browser is finished
+  // writing out all the frame images.
+  // eslint-disable-next-line no-empty
+  let old_mtime = fs.statSync(directory).mtime;
+  await delay(1000);
+  let new_mtime = fs.statSync(directory).mtime;
+  while (old_mtime.getTime() != new_mtime.getTime()) {
+    log.debug(
+      'Still waiting for all frames: ' +
+        old_mtime.getTime() +
+        ' != ' +
+        new_mtime.getTime()
+    );
+    await delay(5000);
+    old_mtime = new_mtime;
+    new_mtime = fs.statSync(directory).mtime;
+  }
+}
+
+async function generateVideo(destination) {
+  let directoryName = findRecordingDirectory();
+  await pollDirectory(directoryName);
+  let imageFiles = [];
+
+  fs.readdirSync(directoryName).forEach(file => {
+    // Format of the filenames are "frame-<num>-<offset>.png"
+    // where num is frame number and offset is time since capture start in ms.
+    let fields = file.split('-');
+    let frameno = ('0000' + fields[1]).slice(-4);
+    let newFilename = 'frame' + frameno + '.png';
+    let offset = fields[2].split('.')[0];
+    fs.renameSync(
+      path.join(directoryName, file),
+      path.join(directoryName, newFilename)
+    );
+    imageFiles.push({ filename: newFilename, offset: offset });
+  });
+
+  imageFiles.sort(function(a, b) {
+    if (a.filename < b.filename) return -1;
+    if (a.filename > b.filename) return 1;
+    return 0;
+  });
+
+  //First step is merging all of the png frames into a single CFR video.
+  const cfr_args = [
+    '-i',
+    path.join(directoryName,'frame%04d.png'),
+    '-vf',
+    'pad=ceil(iw/2)*2:ceil(ih/2)*2',
+    '-pix_fmt',
+    'yuv420p',
+    path.join(directoryName,'tmp-cfr.mp4')
+  ];
+  log.debug('Executing command: ffmpeg ' + cfr_args.join(' '));
+  await execa('ffmpeg', cfr_args);
```

Comment:
Wish we weren't taking `ffmpeg` from the path, but we'll clean this up at some point in the future.  (Gotta do it for `visualmetrics.py` too.)